### PR TITLE
AP_HAL_ChibiOS:add board support for CrazyF405

### PR DESCRIPTION
**Pull request by Srivenkateshwar**
Hardware Description file for CrazyF405.

### CODE REVIEW

**Comment by Srivenkateshwar** [comment]
@Hwurzburg i have created a clean commit you can review this and merge it.Thanks

**Comment by Hwurzburg** [comment]
its failing since you have not created a bootloader and placed in Tools/bootloaders

**Comment by Srivenkateshwar** [comment]
> its failing since you have not created a bootloader and placed in Tools/bootloaders

Okk so according to previous conversation you said you can put bootloader in seperate commit right?

**Comment by Hwurzburg** [comment]
yes...Tools: add CrazyF405 bootloader

**Comment by Hwurzburg** [comment]
while you are at it change the bootloader files APJ_BOARD_ID to text

**Comment by Srivenkateshwar** [comment]
> while you are at it change the bootloader files APJ_BOARD_ID to text

Yeah I have done that in this commit

File: libraries/AP_HAL_ChibiOS/hwdef/CrazyF405/hwdef-bl.dat [hunk 1]
```diff
@@ -0,0 +1,43 @@
+
+# hw definition file for processing by chibios_hwdef.py
+# for BETAFLIGHTF4 hardware.
+# thanks to betaflight for pin information
+
+# MCU class and specific type
+MCU STM32F4xx STM32F405xx
+
+# board ID for firmware load
+APJ_BOARD_ID 1177
```
Comment (Hwurzburg) [inline libraries/AP_HAL_ChibiOS/hwdef/CrazyF405/hwdef-bl.dat:10]: no, you did not change it to text.... It still needs the CrazyF405_bl.bin file added to Tools/bootloaders in order to be able to build the code and pass CI

**Review by Hwurzburg — CHANGES_REQUESTED** [review]

File: libraries/AP_HAL_ChibiOS/hwdef/CrazyF405/hwdef-bl.dat (hunk 1 above)
Comment (Srivenkateshwar) [reply libraries/AP_HAL_ChibiOS/hwdef/CrazyF405/hwdef-bl.dat]: Thanks! I've made the correction, but one check, configure, is still failing. If I'm not mistaken, this check verifies whether the bootloader file is available. I have a separate commit for the bootloader. I believe if you merge that first and then rerun the checks, they should pass successfully.

**Comment by Srivenkateshwar** [comment]
@Hwurzburg some check are still unsuccessful

**Comment by Hwurzburg** [comment]
I restarted CI and they all passed...fixed the commit names 
if it has been tested on the board then I will mark for devcall review and merge

**Comment by Srivenkateshwar** [comment]
> I restarted CI and they all passed...fixed the commit names if it has been tested on the board then I will mark for devcall review and merge

Thanks, yeah it has been tested on the board in stabilize mode. Also need your guidance on more contribution to ardupilot firmware

**Review by Hwurzburg — APPROVED** [review]

**Comment by Hwurzburg** [comment]
@peterbarker noticed that you are using the "include ../include/minimize_common.inc" rather than the preferred for F405 FPV boards "include ../include/minimize_fpv.inc" which will allow the board not to run out of flash longer as AP evolves....if you do not actually need the features removed by it, we should use it

**Comment by Hwurzburg** [comment]
@Srivenkateshwar response? will have to remove the devcall label until we can get a response...

**Comment by Srivenkateshwar** [comment]
> @peterbarker noticed that you are using the "include ../include/minimize_common.inc" rather than the preferred for F405 FPV boards "include ../include/minimize_fpv.inc" which will allow the board not to run out of flash longer as AP evolves....if you do not actually need the features removed by it, we should use it

Okk sorry for late response. We can use minimize fpv.

**Comment by Srivenkateshwar** [comment]
> > @peterbarker noticed that you are using the "include ../include/minimize_common.inc" rather than the preferred for F405 FPV boards "include ../include/minimize_fpv.inc" which will allow the board not to run out of flash longer as AP evolves....if you do not actually need the features removed by it, we should use it
> 
> Okk sorry for late response. We can use minimize fpv.

DO I need to update that ,

**Comment by Hwurzburg** [comment]
I'll fix it for you

**Comment by Srivenkateshwar** [comment]
> I'll fix it for you

Thanks 👍

**Comment by Srivenkateshwar** [comment]
@Hwurzburg can you update a label for this pr??

**Comment by peterbarker** [comment]
Merged, thanks!

**Pull request merged**